lib/puppet/module/task.rb

```diff
@@ -61,6 +62,66 @@ def self.is_tasks_filename?(path)
       return true
     end
 
+    def self.get_file_details(path, name_start)
+      unless File.absolute_path(path) == File.path(path)
+        msg = _("File pathnames cannot include relative paths")
+        raise InvalidMetadata.new(msg, 'puppet.tasks/invalid-metadata')
+      end
+
+      # This gets the path from the starting point onward
+      # For files this should be the file subpath from the metadata
+      # For directories it should be the directory subpath plus whatever we globbed
+      # Partition matches on the first instance it finds of the parameter
+      name = path.partition(name_start).drop(1).join("")
```

Comment:
Did my suggestion not work? This could fail a scenario like `/etc/puppetlabs/code/environments/module_foo/modules/module_foo/...`.

Reply:
I added a test for that scenario and it passes. Partition matches on the first instance of the string, and no others.

Reply:
I don't think your test covers the scenario I just outlined. My scenario is more difficult to test, because it requires the environment or earlier parts of the path to use the module name.